zebedee-reader/src/main/java/com/github/onsdigital/zebedee/util/URIUtils.java

```diff
@@ -79,11 +83,93 @@ public static String getLastSegment(String uri) {
         return segments[segments.length - 1];
     }
 
-    public static String removeLastSegment(String uri) {
-        if (isEmpty(uri)) {
-            return null;
+    public static String removeAfterKeyword(String path, String keyword) {
+        int index = path.indexOf(keyword);
+        if (index == -1) {
+            return path;
+        }
+        String result = path.substring(0, index);
+        return result.endsWith("/") ? result.substring(0, result.length() - 1) : result;
+    }
+
+    public static String getQueryParameterFromURL(String url, String queryParam) {
+        try {
+            URI uri = new URI(StringUtils.lowerCase(url));
+            String query = uri.getQuery();
+            if (query != null && query.contains(StringUtils.lowerCase(queryParam))) {
+                Map<String, String> queryParameterMap = getParameterMapFromQuery(query);
+                String paramValue = queryParameterMap.get(StringUtils.lowerCase(queryParam));
+                return getDecodedUrl(paramValue);
+            }
+        } catch (URISyntaxException e) {
+            throw new RuntimeException("Error retrieving query parameter from path", e);
+        }
+        return null;
+    }
+
+    public static String getDecodedUrl(String url) {
+        if (isUUEncoded(url)) {
+            return URLDecoder.decode(url);
+        }
+        return url;
+    }
+
+    public static boolean isUUEncoded(String url) {
+        return url.toLowerCase().contains("%2f");
+    }
+
+    public static boolean isLastSegmentFileExtension(String path) {
+        if ("".equals(path)){
+            return false;
+        }
+        String lastSegment = getLastSegment(path);
+        if (lastSegment == null){
+            return false;
         }
-        String result =  removeEnd(removeTrailingSlash(uri), FORWARD_SLASH + getLastSegment(uri));
+        return lastSegment.contains(".");
+    }
+
+    public static boolean hasFileExtension(String path) {
```

Comment:
Was this meant to be used in the function above? At the moment it's only used in the tests.

Reply:
removed